### PR TITLE
To only use id after get coll id by name

### PR DESCRIPTION
**Pull request by smellthemoon**
/kind bug
#25662

### CODE REVIEW

File: internal/proxy/impl.go [hunk 1]
```diff
@@ -2619,6 +2619,20 @@ func (node *Proxy) Flush(ctx context.Context, request *milvuspb.FlushRequest) (*
...
+			commonpbutil.WithSourceID(paramtable.GetNodeID()),
+		),
+		ReqID: paramtable.GetNodeID(),
+	}
+	return node.QueryImpl(ctx, request, retrieveRequest)
+}
+
+// QueryImpl implements Query
+// can specific the collection id or other params by pass retrieveRequest
+func (node *Proxy) QueryImpl(ctx context.Context, request *milvuspb.QueryRequest, retrieveRequest *internalpb.RetrieveRequest) (*milvuspb.QueryResults, error) {
```
Comment (yah01) [inline internal/proxy/impl.go:2634]: make this private, and you don't need to modify `types.go` and mock it

Comment (smellthemoon) [reply internal/proxy/impl.go:2634]: mock here to make *MockProxy return QueryImpl resp to test the specific case.

**Pull request closed without merging**